docs/ref/clvm.md

```diff
@@ -603,15 +603,139 @@ When used in an integer context, nil behaves as zero.
 
 To determine the total cost of a clvm program, you can run `brun -c <clvm>`.
 
-This section will explain how the minimum and maximum costs were derived, the maximum size in KB of a CLVM program, the cost of each individual CLVM operator, and how to calculate the total cost of a small sample program.
+This section begins with a breakdown of the specific cost for each operator and how to calculate costs by hand.
 
-Before we can get into those details, it's important to outline the rationale for having costs in the first place. We'll begin with the lowest common denominator in Chia farming: the humble Raspberry Pi 4.
+* [Cost tables](#cost-tables)
+* [Evaluating cost for a sample brun program](#evaluating-cost-for-a-sample-brun-program)
+
+Later, we'll discuss our rationale for having costs in the first place. We'll also detail the theoretical and realistic maximum cost and size per block. 
+
+* [Lowest common denominator to stay synced](#lowest-common-denominator-to-stay-synced)
+* [Maximum cost per block rationale](#maximum-cost-per-block-rationale)
+
+## Cost tables
+
+The costs used in Chia's consensus come from the Rust implementation of CLVM, specifically from these locations:
+  * [more_ops.rs#L24](https://github.com/Chia-Network/clvm_rs/blob/main/src/more_ops.rs#L24)
+  * [core_ops.rs#L7](https://github.com/Chia-Network/clvm_rs/blob/main/src/core_ops.rs#L7)
+  * [run_program.rs#L11](https://github.com/Chia-Network/clvm_rs/blob/main/src/run_program.rs#L11)
+
+We'll start with a table showing the two costs that must be applied to all operators, namely a mandatory cost and a cost per byte of memory.
+
+| type                   | base cost   | cost per byte |
+| ---------------------- | ----------- | ------------- |
+| mandatory cost         | 1           | -             |
+| `MALLOC_COST_PER_BYTE` | -           | 10            |
+
+The following table shows the cost of each CLVM operator, as well as the cost of the outputted conditions.
+
+| operator            | base cost | cost per arg | cost per byte |
+| ------------------- | --------- | ------------ | ------------- |
+| `f` *first*         | 30        | -            | -             |
+| `i` *if*            | 33        | -            | -             |
+| `c` *cons*          | 50        | -            | -             |
+| `r` *rest*          | 30        | -            | -             |
+| `l` *listp*         | 19        | -            | -             |
+| `q` *quote*         | 20        | -            | -             |
+| `a` *apply*         | 90        | -            | -             |
+| `=`                 | 117       | -            | 1             |
+| `+`                 | 99        | 320          | 3             |
+| `/`                 | 988       | -            | 4             |
+| `*`                 | 92        | 885          | [see here](https://github.com/Chia-Network/clvm_tools/blob/main/costs/README.md#multiplication) |
```

Comment:
at some point in the future I'm hoping we can just record the formula directly in this document rather than having this reference. That document is more of a report of measurements and how the formula was developed.

Reply:
It's a little unclear to me what the exact formula is. Is 92 the base cost, or is it 885? Is there a cost per byte, or is that already built into the formula? Also, are the examples meant to represent very large numbers, where the first one has 1400 bytes both before and after the multiplication sign?

Reply:
yeah, for normal numbers, the linear byte cost is dominant. But since we support arbitrary large integers, it's possible to multiply very large numbers. When doing that, the true cost of computing that product is not linear. The algorithms for multiplying large numbers are actually quite sophisticated, we use a simplified model to charge cost.

We apply a (linear) cost per byte of operands (cost 6 per byte)
And we apply a (quadratic) cost per bytes squared (cost 1/128 per byte)

So, the cost for multiplying two values is:

```
(len(lhs) + len(rhs)) * 6 + (len(lhs) * len(rhs)) / 128
```

Where `len()` counts the number of bytes of the integer, and `lhs` is left-hand-side and `rhs` is right-hand-side operands.